src/Validator/CardDataValidator.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Validator;
+
+use App\Repository\CardRepository;
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+
+class CardDataValidator
+{
+    private $cardRepository;
```

Comment:
typehint

Reply:
try to use php 8 constructor argument promotion :)

Reply:
done